### PR TITLE
Entering/Leaving Water, not Supply/Discharge/Return Water

### PR DESCRIPTION
**Pull request by gtfierro**
Addresses #358 . Deprecation notices ~~in-progress~~ added

### CODE REVIEW

File: bricksrc/definitions.csv [hunk 1]
```diff
@@ -362,11 +342,18 @@ https://brickschema.org/schema/Brick#Energy_Storage_System,A collection of devic
 https://brickschema.org/schema/Brick#Energy_System,"A collection of devices that generates, stores or transports electricity",
 https://brickschema.org/schema/Brick#Energy_Usage_Sensor,Measures the total amount of energy used over some period of time,
 https://brickschema.org/schema/Brick#Energy_Zone,A space or group of spaces that are managed or monitored as one unit for energy purposes,
-https://brickschema.org/schema/Brick#Entering_Water,Water that is entering a piece of equipment or system,
-https://brickschema.org/schema/Brick#Entering_Water_Flow_Sensor,Measures the rate of flow of water entering a piece of equipment or system,
-https://brickschema.org/schema/Brick#Entering_Water_Flow_Setpoint,Sets the target flow rate of entering water,
+https://brickschema.org/schema/Brick#Entering_Water,,
```
Comment (jbkoh) [inline bricksrc/definitions.csv:345]: Why do we remove the definition of `Entering_Water`?

File: bricksrc/definitions.csv [hunk 2]
```diff
@@ -362,11 +342,18 @@ https://brickschema.org/schema/Brick#Energy_Storage_System,A collection of devic
...
-https://brickschema.org/schema/Brick#Entering_Water_Flow_Setpoint,Sets the target flow rate of entering water,
+https://brickschema.org/schema/Brick#Entering_Water,,
+https://brickschema.org/schema/Brick#Entering_Water_Differential_Pressure_Deadband_Setpoint,Sets the size of a deadband of differential pressure of supply water,
+https://brickschema.org/schema/Brick#Entering_Water_Differential_Pressure_Integral_Time_Parameter,,
+https://brickschema.org/schema/Brick#Entering_Water_Differential_Pressure_Proportional_Band_Parameter,,
+https://brickschema.org/schema/Brick#Entering_Water_Flow_Sensor,Measures the rate of flow of hot supply water,
+https://brickschema.org/schema/Brick#Entering_Water_Flow_Setpoint,Sets the flow rate of hot supply water,
+https://brickschema.org/schema/Brick#Entering_Water_Temperature_Alarm,An alarm that indicates the off-normal conditions associated with temperature of the supply water.,
+https://brickschema.org/schema/Brick#Entering_Water_Temperature_Deadband_Setpoint,Sets the size of a deadband of temperature of supply water,
```
Comment (jbkoh) [inline bricksrc/definitions.csv:352]: supply has not been changed to entering in those definitions.

**Comment by jbkoh** [comment]
I'd like to improve the naming convention on some points like "Hot Water Leaving Temperature Sensor", which caused inconsistency in naming. They are not consistent with the rest like "Leaving Water Temperature Sensor". Can we make them all consistent like "Leaving Hot Water Temperature Sensor"? Basically, the class names should be more structured with Substance / Quantity names. That way, we can validate the class names easier with a proper substance hierarchy, and users can find things more easily and relate one to another in the hierarchy.

File: bricksrc/alarm.py [hunk 3]
```diff
@@ -192,22 +192,19 @@
                         "tags": [TAG.Point, TAG.Water, TAG.Temperature, TAG.Alarm],
                         "parents": [BRICK.Water_Alarm],
                         "subclasses": {
-                            "Discharge_Water_Temperature_Alarm": {
```
Comment (jbkoh) [inline bricksrc/alarm.py:195]: I think the idea was to keep deprecated concepts in the hierarchy. Is there a different way to preserve that even if we remove them here?

File: bricksrc/deprecations.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    },
+    BRICK.Supply_Condenser_Water_Flow_Sensor: {
+        "version": "1.3.0",
+        "mitigation_message": "Swapped supply/return for entering/leaving with water-related poitns",
+        "replace_with": BRICK.Entering_Condenser_Water_Flow_Sensor,
+    },
+    BRICK.Heat_Exchanger_Supply_Water_Temperature_Sensor: {
+        "version": "1.3.0",
+        "mitigation_message": "Swapped supply/return for entering/leaving with water-related poitns",
+        # "replace_with": BRICK.Leaving_Condenser_Water,
```
Comment (jbkoh) [inline bricksrc/deprecations.py:97]: Are those commented out intentionally? Just heads up.

File: bricksrc/rules.ttl [hunk 5]
```diff
@@ -106,7 +106,7 @@ CONSTRUCT {
 bsh:DeprecationRule
     a   sh:NodeShape ;
     sh:property [
-        sh:path brick:deprecation ;
+        sh:path (rdf:type brick:deprecation) ;
```
Comment (jbkoh) [inline bricksrc/rules.ttl:109]: Please add more description to this in the PR or as a comment in the code. It seems to be changed from the original deprecation design.

File: bricksrc/definitions.csv [hunk 6]
```diff
@@ -989,26 +954,7 @@ https://brickschema.org/schema/Brick#Supply_Air_Temperature_Setpoint,Temperature
...
-https://brickschema.org/schema/Brick#Supply_Water,,
-https://brickschema.org/schema/Brick#Supply_Water_Differential_Pressure_Deadband_Setpoint,Sets the size of a deadband of differential pressure of supply water,
-https://brickschema.org/schema/Brick#Supply_Water_Differential_Pressure_Integral_Time_Parameter,,
-https://brickschema.org/schema/Brick#Supply_Water_Differential_Pressure_Proportional_Band_Parameter,,
-https://brickschema.org/schema/Brick#Supply_Water_Flow_Sensor,Measures the rate of flow of hot supply water,
-https://brickschema.org/schema/Brick#Supply_Water_Flow_Setpoint,Sets the flow rate of hot supply water,
-https://brickschema.org/schema/Brick#Supply_Water_Temperature_Alarm,An alarm that indicates the off-normal conditions associated with temperature of the supply water.,
-https://brickschema.org/schema/Brick#Supply_Water_Temperature_Deadband_Setpoint,Sets the size of a deadband of temperature of supply water,
-https://brickschema.org/schema/Brick#Supply_Water_Temperature_Integral_Time_Parameter,,
-https://brickschema.org/schema/Brick#Supply_Water_Temperature_Proportional_Band_Parameter,,
```
Comment (jbkoh) [inline bricksrc/definitions.csv:1010]: Where did you add the definition for the new classes?

**Review by jbkoh — CHANGES_REQUESTED** [review]
Skimmed through the changes and look good overall. In addition to the minor comments, two changes that I didn't anticipate are:
- Definitions are mostly removed other than replaced. Is it correct? 
- We previously concluded to keep the deprecated concepts within the hierarchy but not in this code. Is it expected? You said you have some script to generate deprecated concepts, but unsure how you'd implement that without keeping the hierarchy (and they are not in this PR, right?)

File: bricksrc/rules.ttl (hunk 5 above)
Comment (gtfierro) [reply bricksrc/rules.ttl:109]: It just fixes a bug where deprecations warnings would fire if you had the deprecation definitions in the graph --- not when you actually instantiated a deprecated concept

File: bricksrc/definitions.csv (hunk 6 above)
Comment (gtfierro) [reply bricksrc/definitions.csv:1010]: forgot to commit, whoops!

File: bricksrc/alarm.py (hunk 3 above)
Comment (gtfierro) [reply bricksrc/alarm.py:195]: They will be in the deprecation file

**Comment by gtfierro** [comment]
Thanks for the review! All comments have been addressed

**Pull request merged**